### PR TITLE
Add java.util.IdentityHashMap (#1452)

### PR DESCRIPTION
**Pull request by ekrich**
* Added IdentityHashMapSuite, forwarders and fix equals

* Fix to allow default set of tests to run

* Adapt suite to allow reuse with IdentityHashMap

* Fix formatting

* Cleaned up with shared and Harmony tests

* Enough tests for now

* Fix last test by adding parens

### CODE REVIEW

File: javalib/src/main/scala/java/util/IdentityHashMap.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    }
+
+  // forwarders - https://github.com/scala-native/scala-native/issues/375
+  override def containsKey(key: Any): Boolean     = super.containsKey(key)
+  override def containsValue(value: Any): Boolean = super.containsValue(value)
+  override def get(key: Any): V                   = super.get(key)
+  override def put(key: K, value: V): V           = super.put(key, value)
+  override def remove(key: Any): V                = super.remove(key)
+  override def clear(): Unit                      = super.clear()
+  override def size(): Int                        = super.size()
```
Comment (densh) [inline javalib/src/main/scala/java/util/IdentityHashMap.scala]: You shouldn't need this forwarders on master.

Comment (ekrich) [reply javalib/src/main/scala/java/util/IdentityHashMap.scala]: Good, catch. Will fix.

**Comment by densh** [comment]
LGTM! 👍 Just waiting for CI to confirm it's all good now.

**Comment by ekrich** [comment]
Super 👍

**Comment by ekrich** [comment]
Refer to Issue #1412

**Pull request merged**